src/MemoryGraph/graph.cs

```diff
@@ -141,6 +141,10 @@ public virtual NodeType AllocTypeNodeStorage()
         /// </summary>
         public int TotalNumberOfReferences { get { return m_totalRefs; } }
 
+        protected int m_segmentSize;
```

Comment:
❔ Can we inline a constant here? It doesn't seem especially advantageous to make it configurable.

Reply:
I have been using a size of 131_072 and it has worked fine, however this value does affect performance and the amount of fragmentation that occurs, that is why I thought it best to be configurable.

Reply:
We should consider a value which avoids the creation of pages in the Large Object Heap. I believe the largest such value would be 8192.

Reply:
I was looking into this and I found that the LOH threshold is 85,000 bytes (According to [here](https://blogs.msdn.microsoft.com/dotnet/2011/10/03/large-object-heap-improvements-in-net-4-5/)). In that case limiting the size to 8192 elements would make it not enter LOH. Is that what you were referring to? Avoid entering the LOH?